manage.py

```diff
@@ -129,12 +129,57 @@ def set_dev_site_name() -> None:
                 name="pythondiscord.local:8000"
             )
 
+    @staticmethod
+    def run_metricity_init() -> None:
+        """
+        Initilise metricity relations and populate with some testing data.
+
+        This is done at run time since other projects, like Python bot,
+        rely on the site initilising it's own db, since they do not have
+        access to the init.sql file to mount a docker-compose volume.
+        """
+        import psycopg2
+        from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+        from urllib.parse import urlsplit
+        # The database URL has already been validated in `wait_for_postgres()`
+        db_url_parts = urlsplit(os.environ["DATABASE_URL"])
+        db_connection_kwargs = {
+            "host": db_url_parts.hostname,
+            "port": db_url_parts.port,
+            "user": db_url_parts.username,
+            "password": db_url_parts.password,
+        }
+        # Connect to pysite first to create metricity db
+        conn = psycopg2.connect(
+            database=db_url_parts.path[1:],
+            **db_connection_kwargs
+        )
+        conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+
+        with conn.cursor() as cursor:
+            cursor.execute("SELECT 1 FROM pg_catalog.pg_database WHERE datname = 'metricity'")
+            exists = cursor.fetchone()
+            if exists:
+                # Assume metricity is already populated if it exists
+                return
+            print("Creating metricity relations and populating with some data.")
+            cursor.execute("CREATE DATABASE metricity")
+
+        # Switch connection to metricity and initialise some data
+        conn = psycopg2.connect(
```

Comment:
Does the previous connection not need to be cleaned up in any way?

Reply:
ah yes, you're right. I had assumed leaving the context manager dealt with this, but it does not.

Reply:
Resolved here https://github.com/python-discord/site/pull/583/commits/6de87e47e313b4f533e8d3adfeb0b8202e938ca0